test/test_um2netcdf.py

```diff
@@ -1069,13 +1069,29 @@ def test_fix_pressure_levels_reverse_pressure():
 
 # int64 to int32 data conversion tests
 # NB: skip float64 to float32 overflow as float32 min/max is huge: -/+ 3.40e+38
+
+def test_convert_32_bit_safe(ua_plev_cube):
+    # simple baseline test of down conversion int64 to int32 without warnings
+    data = [1e6, 200, 100, 10, 1, 0, -10]
+    ua_plev_cube.data = np.array(data, dtype=np.int64)
+    um2nc.convert_32_bit(ua_plev_cube)
+    assert ua_plev_cube.data.dtype == np.int32
+    assert np.all(ua_plev_cube.data == data)
+
+
 @pytest.mark.parametrize("array,_operator,bound",
-                         [([100, 10, 1, 0, -10], None, None),
-                          ([3000000000], operator.gt, np.iinfo(np.int32).max),
+                         [([3000000000], operator.gt, np.iinfo(np.int32).max),
                           ([-3000000000], operator.lt, np.iinfo(np.int32).min)])
-def test_convert_32_bit(ua_plev_cube, array, _operator, bound):
+def test_convert_32_bit_overflow_warning(ua_plev_cube, array, _operator, bound):
+    # ensure overflow covered for large positive & negative int64s
     ua_plev_cube.data = np.array(array, dtype=np.int64)
-    um2nc.convert_32_bit(ua_plev_cube)
+
+    with pytest.warns(RuntimeWarning) as record:
+        um2nc.convert_32_bit(ua_plev_cube)
+
+        if not record:
+            msg = f"No overflow warning with {array} {_operator} {bound}"
+            pytest.fail(msg)
 
     if _operator:
         assert _operator(array[0], bound)
```

Comment:
I think the separation of the test into the `safe` and `overflow` cases helps make it easier to read! Now that there's no `_operator == None` case in this test, is it worth removing the `if _operator` check?

I'm also wondering, is this check included to make sure the test setup is correct? I'm wondering if would be clearest to have it near the start of the test (e.g. around line 1088)?

Reply:
I've fixed both in 85ef8b6f & cleaned the error messages.

There's a safety net If test params are somehow munged in other commits.